core/metacling/src/TClingCallbacks.cxx

```diff
@@ -783,3 +785,11 @@ void TClingCallbacks::LibraryUnloaded(const void* dyLibHandle,
 void TClingCallbacks::PrintStackTrace() {
    TCling__PrintStackTrace();
 }
+
+void TClingCallbacks::EnteringUserCode() {
+  TCling__UnlockInterpreterMutex();
```

Comment:
Add
```
// We can safely assume that if the lock exist already when we are in Cling code,
// then the lock has (or should been taken) already.  So it is fair to unlock it.
```
However if the mutex is still null here we should record that information has it means we ought not to unlock it on exit ....